perllib/FixMyStreet/Cobrand/Oxfordshire.pm

```diff
@@ -284,4 +289,108 @@ sub dashboard_export_problems_add_columns {
+                <ogc:Literal>With Contractor</ogc:Literal>
+            </ogc:PropertyIsEqualTo>
+            <ogc:BBOX>
+                <ogc:PropertyName>SHAPE_GEOMETRY</ogc:PropertyName>
+                <gml:Envelope xmlns:gml=\"http://www.opengis.net/gml\" srsName=\"$bbox->[4]\">
+                    <gml:lowerCorner>$bbox->[0] $bbox->[1]</gml:lowerCorner>
+                    <gml:upperCorner>$bbox->[2] $bbox->[3]</gml:upperCorner>
+                </gml:Envelope>
+            </ogc:BBOX>
+        </ogc:And>
+    </ogc:Filter>";
+    $filter =~ s/\n\s+//g;
+
+    my $uri = URI->new("https://tilma.mysociety.org/proxy/occ/nsg/");
+    $uri->query_form(
+        REQUEST => "GetFeature",
+        SERVICE => "WFS",
+        SRSNAME => "urn:ogc:def:crs:EPSG::4326",
+        TYPENAME => "WFS_DEFECTS_FOR_QUERYING",
+        VERSION => "1.1.0",
+        filter => $filter,
+        propertyName => 'ITEM_CATEGORY_NAME,ITEM_TYPE_NAME,REQUIRED_COMPLETION_DATE,SHAPE_GEOMETRY',
+        outputformat => "application/json"
+    );
+
+    try {
+        my $response = get($uri);
+        my $json = JSON->new->utf8->allow_nonref;
+        return $json->decode($response);
+    } catch {
+        # Ignore WFS errors.
+        return [];
+    };
+}
+
+# Get defects from WDM feed and display them on /around page.
+sub pins_from_wfs {
+    my ($self, $bbox) = @_;
+
+    my $wfs = $self->defect_wfs_query($bbox);
+
+    # Generate a negative fake ID so it doesn't clash with FMS report IDs.
+    my $fake_id = -1;
+    my @pins = map {
+        my $coords = $_->{geometry}->{coordinates};
+        my $props = $_->{properties};
+        my $category = $props->{ITEM_CATEGORY_NAME};
+        my $type = $props->{ITEM_TYPE_NAME};
+        my $category_type;
+        $category =~ s/\s+$//;
+        $type =~ s/\s+$//;
+        if ($category eq $type) {
+            $category_type = $category;
+        } else {
+            $category_type = "$category ($type)";
+        }
+        my $completion_date = DateTime::Format::W3CDTF->parse_datetime($props->{REQUIRED_COMPLETION_DATE})->strftime('%A %e %B %Y');
+        my $title = "$category_type\nCompletion date: $completion_date";
+        {
+            id => $fake_id--,
+            latitude => @$coords[1],
+            longitude => @$coords[0],
+            colour => 'defects',
+            title => $title,
+        };
+    } @{ $wfs->{features} };
+
+    return \@pins;
+}
+
+sub extra_nearby_pins {
+    my ($self, $latitude, $longitude, $dist) = @_;
+
+    my ($easting, $northing) = Utils::convert_latlon_to_en($latitude, $longitude);
+    my $bbox = [$easting-$dist, $northing-$dist, $easting+$dist, $northing+$dist, 'EPSG:27700'];
+
+    my $pins = $self->pins_from_wfs($bbox);
+
+    return map {
+        [ $_->{latitude}, $_->{longitude}, $_->{colour},
+          $_->{id}, $_->{title}, "normal", JSON->false
+        ]
+    } @$pins;
+}
+
+sub extra_around_pins {
+    my ($self, $bbox) = @_;
```

Comment:
`$bbox` might be `undef` if the user isn't using JS (e.g. they've gone to a URL like `/around?pc=witney` without the `js=1`param), so we should return in that case to prevent an invalid request to the WFS server (e.g. https://tilma.mysociety.org/proxy/occ/nsg/?REQUEST=GetFeature&SERVICE=WFS&SRSNAME=urn%3Aogc%3Adef%3Acrs%3AEPSG%3A%3A4326&TYPENAME=WFS_DEFECTS_FOR_QUERYING&VERSION=1.1.0&filter=%3Cogc%3AFilter+xmlns%3Aogc%3D%22http%3A%2F%2Fwww.opengis.net%2Fogc%22%3E%3Cogc%3AAnd%3E%3Cogc%3APropertyIsEqualTo+matchCase%3D%22true%22%3E%3Cogc%3APropertyName%3EAPPROVAL_STATUS_NAME%3C%2Fogc%3APropertyName%3E%3Cogc%3ALiteral%3EWith+Contractor%3C%2Fogc%3ALiteral%3E%3C%2Fogc%3APropertyIsEqualTo%3E%3Cogc%3ABBOX%3E%3Cogc%3APropertyName%3ESHAPE_GEOMETRY%3C%2Fogc%3APropertyName%3E%3Cgml%3AEnvelope+xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%22+srsName%3D%22%22%3E%3Cgml%3AlowerCorner%3EEPSG%3A4326+%3C%2Fgml%3AlowerCorner%3E%3Cgml%3AupperCorner%3E+%3C%2Fgml%3AupperCorner%3E%3C%2Fgml%3AEnvelope%3E%3C%2Fogc%3ABBOX%3E%3C%2Fogc%3AAnd%3E%3C%2Fogc%3AFilter%3E&propertyName=ITEM_CATEGORY_NAME%2CITEM_TYPE_NAME%2CREQUIRED_COMPLETION_DATE%2CSHAPE_GEOMETRY&outputformat=application%2Fjson)

Reply:
Ah good catch. Fixed in 88bd91ec9dad55b0553230ce8de062c18d209ab2